hack/shellcheck-test.sh

```diff
@@ -27,35 +29,38 @@ BASE_DIR="$(cd "${SCRIPT_DIR}/.." && pwd)"
 if [[ -z "${SHELLCHECK}" ]]; then
         echo "warning: could not find shellcheck ... installing shellcheck" >&2
         scversion="stable"
+        FILE="${OUTDIR_TOOLS}/shellcheck"
         wget -qO- "https://storage.googleapis.com/shellcheck/shellcheck-${scversion?}.linux.x86_64.tar.xz" | tar -xJv
-        cp "shellcheck-${scversion}/shellcheck" "$GOPATH"/src/github.com/openshift/ocs-operator/tools/
-        FILE="$GOPATH"/src/github.com/openshift/ocs-operator/tools/shellcheck
+        cp -f "shellcheck-${scversion}/shellcheck" "${FILE}"
         if [ -f "$FILE" ]; then
                 SHELLCHECK="${FILE}"
         fi
 fi
 
 cd "${BASE_DIR}" || exit 2
-SCRIPTS=$(find "$GOPATH"/src/github.com/openshift/ocs-operator \( -path "*/vendor/*" -o -path "*/build/*" -o -path "*/_cache/*" \) -prune -o -name "*~" -prune -o -type f -exec grep -l -e '^#!/bin/bash$' {} \;)
+SCRIPTS=$(find . \( -path "*/vendor/*" -o -path "*/build/*" -o -path "*/_cache/*" \) -prune -o -name "*~" -prune -o -name "*.swp" -prune -o -type f -exec grep -l -e '^#!/bin/bash$' {} \;)
 
 failed=0
 for script in ${SCRIPTS}; do
         err=0
         test_syntax "${script}"
-        if [[ ${err} -ne 0 ]]; then
+        if [[ $? -ne 0 ]]; then
                 err=1
                 echo "detected syntax issues in ${script}}" >&2
         fi
         test_shellcheck "${script}"
-        if [[ ${err} -ne 0 ]]; then
+        if [[ $? -ne 0 ]]; then
                 err=1
                 echo "detected shellcheck issues in ${script}" >&2
         fi
         if [[ $err -ne 0 ]]; then
-                ((failed=err))
+                ((failed+=err))
         else
                 echo "${script}: ok" >&2
         fi
 done
+
+echo "${failed} scripts with errors were found"
+
 exit ${failed}
```

Comment:
do we want to exit the script with number of files failed with errors instead of 1

Reply:
yes. !=0 is error

Reply:
cool